hudi-common/src/main/java/org/apache/hudi/metadata/BaseTableMetadata.java

```diff
@@ -275,30 +243,51 @@ public Map<Pair<String, String>, HoodieMetadataColumnStats> getColumnStats(final
 
     List<String> columnStatKeys = new ArrayList<>(sortedKeys);
     HoodieTimer timer = HoodieTimer.start();
-    List<Pair<String, Option<HoodieRecord<HoodieMetadataPayload>>>> hoodieRecordList =
+    Map<String, HoodieRecord<HoodieMetadataPayload>> hoodieRecords =
```

Comment:
The eventual sorting happens deep down within the HoodieHFileDataBlock
    // HFile read will be efficient if keys are sorted, since on storage records are sorted by key.
    // This will avoid unnecessary seeks.
    List<String> sortedKeys = new ArrayList<>(keys);
    Collections.sort(sortedKeys);

I think at BaseTableMetadata we should not assume the underlying implementation is HFile and hence sorting keys is faster. We should leave that to the storage block.

When multiple shards are involved, the keys here would be split into multiple smaller key ranges with each range looked up on a single shard. So sorting within that smaller range is all we require. For this also it makes sense to leave the sorting to the lower storage block layer.

Reply:
There are other code paths (e.g. BaseTableMetadata::fetchAllFilesInPartitionPaths) where the keys are not sorted before calling getRecordByKeys(). It will be hard to require all code paths to sort before calling.

Reply:
PTAL: I have removed the sorting at BaseTableMetadata.